html_css/v2/main-course/intermediate_css/frameworks-preprocessors.md

```diff
@@ -0,0 +1,91 @@
+### Introduction
+
+At this point, you have written quite a bit of vanilla HTML and CSS, and learned many design techniques that you will continue to use as you grow as a developer. However, through your experiences, you have no doubt encountered a CSS framework or two. You may have also learned about preprocessors (aka precompilers). Both of these types of tools can make writing CSS more streamlined and less tedious.
+
+You should be aware that at this point in your learning, both frameworks and preprocessors are options for you, but neither of them are necessary. In particular, you'll read a strong argument below AGAINST using frameworks at this point.
+
+### Learning Outcomes
+
+- What are CSS frameworks?
+- How do frameworks actually work to get you the look you want?
+- How do you interact with the framework to get the look you want?
+- What are Bootstrap, Foundation, Tailwind and Bulma?
+- What other frameworks are find out there?
+- What are the disadvantages of using frameworks?
+
+- What is a preprocessor?
+- What are the advantages of using a preprocessor?
+- What does it mean that a preprocessor "compiles to CSS"?
+- What is SASS/SCSS?
+- How can you use SASS to nest CSS?
+- How can you use SASS to work with variables in your code?
+- What are the advantages of using mix-ins with SASS?
+- If you're not using Rails, how do you set up SASS?
+- What is LESS and how is it different from SASS?
+- What other preprocessors are out there?
+
+### Frameworks Overview
+
+Frameworks like [Bootstrap]](https://getbootstrap.com/) and [Foundation](https://get.foundation) do a lot of the heavy lifting of packaging up commonly used CSS code, even icons and interactions (like menu dropdowns), for you. They are designed to abstract away the process of coding intuitive, reusable, and responsive elements. They do this by creating these elements for you, and giving you the tools to inject styles into your HTML elements directly through class name keywords. All you have to do is understand how a framework expects you to lay out your site and which classes it uses to designate its particular batch of styles.
+
+While the two frameworks mentioned above are heavy hitters that have been around for a while, you should be aware that there are quite a few others at this point, as evidenced by all of the lists of popular frameworks on the web. Two other extremely popular frameworks are [Tailwind](https://tailwindcss.com/) and [Bulma](https://bulma.io/). There are many more out there.
```

Comment:
i wouldn't call foundation a heavy hitter anymore lol.   THis doesn't matter too much.. but instead of trying to rank popularity I'd say something like 

> "there are a ton of frameworks for you to choose from, in addition to the ones mentioned above......."

Reply:
OK